layers/meta-resin-generic/recipes-bsp/grub/grub_%.bbappend

```diff
@@ -0,0 +1,44 @@
+FILESEXTRAPATHS_prepend := "${THISDIR}/files:"
+
+SRC_URI_append = " \
+    file://config \
+    file://grub.cfg_external \
+    file://grub.cfg_internal-dev \
+    file://grub.cfg_internal-prod \
+    "
+
+inherit deploy
+
+DEPENDS_class-target = "grub-native"
+RDEPENDS_${PN}_class-native = ""
+
+GRUB_IMAGE = "core.img"
+GRUB_TARGET = "i386-pc"
+
+GRUB_BUILDIN ?= "biosdisk part_msdos fat search linux"
+
+do_deploy() {
+    mkdir -p ${DEPLOYDIR}/grub/${GRUB_TARGET}
+    install -m 644 ${D}/${libdir}/grub/${GRUB_TARGET}/boot.img ${DEPLOYDIR}/grub
+    install -m 644 ${WORKDIR}/grub.cfg_external ${DEPLOYDIR}/grub/grub.cfg
+
+    grub-mkimage -c ../config -p /grub -d ./grub-core/ -O ${GRUB_TARGET} \
+                 -o ${DEPLOYDIR}/grub/${GRUB_IMAGE} ${GRUB_BUILDIN}
+
+    if ${@bb.utils.contains('DISTRO_FEATURES','development-image','true','false',d)}; then
+        install -m 644 ${WORKDIR}/grub.cfg_internal-dev ${DEPLOYDIR}/grub.cfg_internal
+    else
+        install -m 644 ${WORKDIR}/grub.cfg_internal-prod ${DEPLOYDIR}/grub.cfg_internal
+    fi
+
+    mkdir -p ${DEPLOYDIR}/grub/${GRUB_TARGET}
+    cp -r ${D}/${libdir}/grub/${GRUB_TARGET}/*.mod ${DEPLOYDIR}/grub/${GRUB_TARGET}
+}
+
+do_deploy_class-native() {
```

Comment:
I think you can avoid this by defining the above function as `do_deploy_class-target`. In this way you don't need to define anything as do_deploy_class-native().

Reply:
But do_deploy_class-native would still be defined and empty => in bitbake complaining

Reply:
Would it?